mods/ra/maps/soviet-08a/soviet08a.lua

```diff
@@ -0,0 +1,160 @@
+		entryPath = { EastEntry.Location, EastUnload.Location + CVec.New(1, 0) },
+		exitPath = { EastEntry.Location },
+	},
+	south = {
+		actors = { "e4", "e4", "e1", "e1", "e1" },
+		entryPath = { SouthEntry.Location, SouthUnload.Location + CVec.New(0, 1) },
+		exitPath = { SouthEntry.Location }
+	},
+	mammoth = {
+		actors = { "4tnk" },
+		entryPath = { SouthEntry.Location, SouthUnload.Location + CVec.New(0, 1) },
+		exitPath = { SouthEntry.Location }
+	}
+}
+
+Obj2ActorTriggerActivator = { Actor146, Actor147, Actor148, Actor149, Actor150, Actor151, Actor152, Actor153 }
+
+AddEastReinforcementTrigger = function()
+	Trigger.AfterDelay(DateTime.Seconds(30), function()
+		Media.PlaySpeechNotification(ussr, "ReinforcementsArrived")
+		local reinforcement = ussrReinforcements.east
+		Reinforcements.ReinforceWithTransport(ussr, "lst.reinforcement", reinforcement.actors, reinforcement.entryPath, reinforcement.exitPath)
+	end)
+end
+
+AddSouthReinforcementTrigger = function()
+	Trigger.AfterDelay(DateTime.Seconds(60), function()
+		Media.PlaySpeechNotification(ussr, "ReinforcementsArrived")
+		local reinforcement = ussrReinforcements.south
+		Reinforcements.ReinforceWithTransport(ussr, "lst.reinforcement", reinforcement.actors, reinforcement.entryPath, reinforcement.exitPath)
+	end)
+end
+
+AddParadropReinforcementTrigger = function()
+	Trigger.AfterDelay(DateTime.Seconds(90), function()
+		Media.PlaySpeechNotification(ussr, "ReinforcementsArrived")
+		powerproxy.SendParatroopers(waypoint35.CenterPosition, false, 10)
+	end)
+end
+
+ChurchAmbushTrigger = function()
+	if not AmbushSwitch then
+		local hiding = Reinforcements.Reinforce(germany, { 'e1', 'e1', 'e1', 'e1', 'e1', 'e1', 'e1', 'e3', 'e3', 'e3' }, { waypoint78.Location, waypoint5.Location })
+		Utils.Do(hiding, function(actor)
+			IdleHunt(actor)
+		end)
+	end
+		AmbushSwitch = true
+end
+
+Obj2TriggerFunction = function()
+	ussr.MarkCompletedObjective(DestroyVillageObjective)
+	Media.PlaySpeechNotification(ussr, "ReinforcementsArrived")
+	local reinforcement = ussrReinforcements.mammoth
+	Reinforcements.ReinforceWithTransport(ussr, "lst.reinforcement", reinforcement.actors, reinforcement.entryPath, reinforcement.exitPath)
+end
+
+AddReinforcmentTriggers = function()
+	AddEastReinforcementTrigger()
+	AddSouthReinforcementTrigger()
+	AddParadropReinforcementTrigger()
+end
+
+AddRetreatTrigger = function()
+	Trigger.OnEnteredProximityTrigger(Actor222.CenterPosition, WDist.FromCells(12), function(actor, id)
+		if (actor.Owner == ussr and actor.Type == "barr") then
+			alliedScouts = Utils.Where(alliedScouts, function(scout) return not scout.IsDead end)
+			for i, scout in ipairs(alliedScouts) do
+				if (scout.Type == "e1") then
+					table.remove(alliedScouts, i)
+					break
+				end
+			end
+			for i, scout in ipairs(alliedScouts) do
+				scout.Stop()
+				scout.Move(waypoint1.Location, 1)
+			end
+			Trigger.RemoveProximityTrigger(id)
+		end
+	end)
+end
+
+Tick = function()
+	greece.Cash = 1000
```

Comment:
This feels like we're hiding some other issue. What exactly was the reason again why the AI to runs out of money? Do the harvesters get stuck?

Reply:
I just figured there's no way AI can keep up constant infantry and vehicle production off of one harvester mining long-distance. That takes four in competitive play.

The mission Evacuation does Cash = 1000 as well.

Reply:
I really don't like cheating AI, but I guess if we didn't this mission would be too easy :/

Reply:
The map doesn't seem to have ore mines so the trucks run low soon unless you expand.